hooks/tk-multi-loader2/tk-nuke_actions.py

```diff
@@ -228,6 +230,46 @@ def _create_read_node(self, path, sg_publish_data):
             # override the detected frame range.
             read_node["first"].setValue(seq_range[0])
             read_node["last"].setValue(seq_range[1])
+        else:
+            self.parent.logger.warning("{}: Not setting frame range.".format(read_node.name()))
+
+        # try to fetch a proxy path using templates
+        proxy_path = self._get_proxy_path(path)
+
+        if proxy_path:
+            read_node["proxy"].fromUserText(proxy_path)
+        else:
+            self.parent.logger.warning("{}: Not setting proxy path.".format(read_node.name()))
+
+    def _get_proxy_path(self, path):
+        # find a template that matches the path:
+        template = None
+        try:
+            template = self.parent.sgtk.template_from_path(path)
+        except sgtk.TankError:
+            pass
+        if not template:
+            return None
+
+        # get the fields
+        fields = template.get_fields(path)
+
+        # get proxy template
+        proxy_template_exp = "{env_name}_proxy_image"
```

Comment:
We should not be hardcoding template names in code. We should instead reference environment settings in code, and associate those settings with a template.

Reply:
By environment settings, you mean app settings? If yes, we haven't forked tk-multi-loader2 and no existing setting seems to be usable for this purpose. Should we fork that as well?

Reply:
Currently, we are creating proxies as a by-product of review submission (since that nuke graph is easily manipulated) Because of this, proxies do not get published. This is probably not the right place to be creating them.
What do you suggest we do instead? Should we write a "create_proxy" plugin for Shotgun Ingest?

Reply:
These proxies are only used by Roto department to track their elements and make scripts less heavy, and none of these gets delivered to the client or used by any other departments. Since the compression on these images is too heavy to be used in Paint or Green Screen Removal by Comp. Do we really need to push these out in a separate PublishedFile entity knowing all this?

Reply:
create_proxy could just be a flag on the ingest_files plugin rather than its own plugin, and could be passed as an argument to tk-multi-reviewsubmission to later be used by the nuke pre-process hook, so that it can be executed in the same nuke subprocess. There should be support in tk-multi-reviewsubmission for arbitrary parameters via *args and **kwargs all the way through to the hook.

Yes, we should publish the proxies since they are still inputs that are used in the production of shot data and that should be tracked. As an added incentive, this aligns with how they are treated in the DD pipeline.